src/lib/services/validatorService.ts

```diff
@@ -192,15 +192,22 @@ export const useValidatorDelegationRelatedTxs = (
 export const useValidatorProposedBlocks = (
   validatorAddress: ValidatorAddr,
   limit: number,
-  offset: number
+  offset: number,
+  options: Pick<UseQueryOptions<BlocksResponse>, "onSuccess"> = {}
 ) => {
   const endpoint = useBaseApiRoute("validators");
 
   return useQuery<BlocksResponse>(
-    [CELATONE_QUERY_KEYS.VALIDATOR_PROPOSED_BLOCKS, endpoint, validatorAddress],
+    [
+      CELATONE_QUERY_KEYS.VALIDATOR_PROPOSED_BLOCKS,
+      endpoint,
+      validatorAddress,
+      limit,
+      offset,
+    ],
     async () =>
       getValidatorProposedBlocks(endpoint, validatorAddress, limit, offset),
-    { retry: 1 }
+    { ...options, retry: 1 }
```

Comment:
let's do `{ retry: 1, ...options }` in case in the future we want to be able to pass custom retry, so we can overwrite `retry: 1`